resolvers.py

```diff
@@ -19,11 +21,62 @@ def get_dates_and_price_check(
             **search_criteria
     ) -> List[DatesCheck]:
         validated_search_criteria = DatesAndPriceCheckInputValidator(**search_criteria)
-        results_list = []
+        houses_list = {}
         for listing_id in validated_search_criteria.listing_ids:
-            listing = data[listing_id]
-            dates_check = DatesCheck(
-                listing_id=listing.listing_id
-            )
-            results_list.append(dates_check)
+            houses_list[listing_id] = data[listing_id]
+        return QueryResolvers.getResultsList(validated_search_criteria, houses_list)
+
+    @staticmethod
+    @query.field("ranked_search_results")
+    def get_ranked_search_results(
+            _,
+            info,
+            **search_criteria
+    ) -> List[DatesCheck]:
+        validated_search_criteria = RankedSearchResultInputValidator(**search_criteria)
+        return QueryResolvers.getResultsList(validated_search_criteria, data)
+
+    @staticmethod
+    def getResultsList(validated_search_criteria, houses_list: dict) -> List[DatesCheck]:
+        results_list: List[DatesCheck] = []
+        checkin_Desired = validated_search_criteria.desired_checkin
+        checkout_Desired = validated_search_criteria.desired_checkout
+        if checkin_Desired is None and checkout_Desired is None:
+            for house in houses_list.values():
+                data_check = DatesCheck(house.listing_id, True, house.available_ranges[0].range_start,
```

Comment:
Hmm this logic is not fully correct. Let's say we have an available range that has already started, for e.g. `range_start=2020-12-01, range_end=2021-05-01`. Then your method will choose `checkin_date=2020-12-01` which is in the past.